src/app/myworkflows/my-workflow/my-workflow.component.ts

```diff
@@ -61,91 +65,214 @@ export class MyWorkflowComponent implements OnInit {
-      this.workflows = workflows;
-      if (this.user) {
+    this.workflowService.workflows$.takeUntil(this.ngUnsubscribe).subscribe(workflows => {
+      if (workflows) {
+        this.workflows = workflows;
         const sortedWorkflows = this.myworkflowService.sortORGWorkflows(workflows, this.user.username);
-        this.workflowService.setNsWorkflows(sortedWorkflows);
-      }
-    });
-    this.workflowService.nsWorkflows$.subscribe(nsWorkflows => {
-      this.orgWorkflows = nsWorkflows;
-      if (this.orgWorkflows && this.orgWorkflows.length > 0) {
-        const foundWorkflow = this.findWorkflowFromPath(this.urlResolverService.getEntryPathFromUrl(), this.orgWorkflows);
-        const theFirstWorkflow = this.orgWorkflows[0].workflows[0];
-        if (foundWorkflow) {
-          this.selectWorkflow(foundWorkflow);
+        /* For the first initial time, set the first tool to be the selected one */
+        if (sortedWorkflows && sortedWorkflows.length > 0) {
+          this.orgWorkflowsObject = this.convertNamespaceWorkflowsToOrgWorkflowsObject(sortedWorkflows);
+          const foundWorkflow = this.findWorkflowFromPath(this.urlResolverService.getEntryPathFromUrl(), this.orgWorkflowsObject);
+          if (foundWorkflow) {
+            this.selectWorkflow(foundWorkflow);
+          } else {
+            const publishedWorkflow = this.getFirstPublishedWorkflow(sortedWorkflows);
+            if (publishedWorkflow) {
+              this.selectWorkflow(publishedWorkflow);
+            } else {
+              const theFirstWorkflow = sortedWorkflows[0].workflows[0];
+              this.selectWorkflow(theFirstWorkflow);
+            }
+          }
         } else {
-          this.selectWorkflow(theFirstWorkflow);
+          this.selectWorkflow(null);
         }
-      } else {
-        this.selectWorkflow(null);
       }
     });
     this.stateService.refreshMessage$.subscribe(refreshMessage => this.refreshMessage = refreshMessage);
   }
 
-  private goToWorkflow(workflow: ExtendedWorkflow): void {
-    this.workflowService.setWorkflow(workflow);
-    this.router.navigateByUrl('/my-workflows/' + workflow.full_workflow_path);
+  ngOnDestroy() {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
   }
 
-  private findWorkflowFromPath(path: string, orgWorkflows: any[]): ExtendedWorkflow {
-    let matchingWorkflow: ExtendedWorkflow;
-    orgWorkflows.forEach((orgWorkflow)  => {
-      orgWorkflow.workflows.forEach(workflow => {
-        if (workflow.full_workflow_path === path) {
-          matchingWorkflow = workflow;
+  /**
+   * This figures out which tab (Published/Unpublished) is active
+   * In order of priority:
+   * 1. If the selected entry is published/unpublished, the tab selected will published/unpublished to reflect it
+   * 2. If there are published entries, the published tab will be selected
+   * 3. Unpublished otherwise
+   * @private
+   * @memberof MyWorkflowComponent
+   */
+  private updateActiveTab() {
```

Comment:
Structure here seems like a copy-paste of https://github.com/dockstore/dockstore-ui2/pull/250/files#diff-bb18135c061ae38bef3adf1efab8d9dfR129 with only types changed.